src/Illuminate/Events/Dispatcher.php

```diff
@@ -156,19 +156,31 @@ public function flush($event)
      *
      * @param  object|string  $subscriber
      * @return void
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
      */
     public function subscribe($subscriber)
     {
         $subscriber = $this->resolveSubscriber($subscriber);
 
-        $subscriber->subscribe($this);
+        $events = $subscriber->subscribe($this);
+
+        if (is_array($events)) {
+            foreach ($events as $event => $listeners) {
+                foreach (array_unique($listeners) as $listener) {
+                    $this->listen($event, $listener);
+                }
+            }
+        }
     }
 
     /**
      * Resolve the subscriber instance.
      *
      * @param  object|string  $subscriber
      * @return mixed
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
```

Comment:
This method doesn't throws this exception, please remove it.

Reply:
Why did you re-add this?

Reply:
I asked about the protected method. you said yeah.
 I think it was a misunderstanding. Will remove it in a moment, sorry.

```
So the protected method which calls the container->make() should have the @throw? if I got it right
In this case resolveSubscriber
```